src/OpenTelemetry.Exporter.Prometheus/Implementation/PrometheusExporterMiddleware.cs

```diff
@@ -79,7 +79,9 @@ public async Task InvokeAsync(HttpContext httpContext)
                     }
                     else
                     {
-                        throw new InvalidOperationException("Collection failure.");
+                        // It's not expected to have no metrics to collect, but it's not necessarily a failure, either.
```

Comment:
Could you do the same logic for the "HttpListener" as well? `PrometheusExporterHttpServer` would also be having this same issue.

Reply:
Added the same logic to `PrometheusExporterHttpServer` along with associated test. Refactored test fixture to be closer to what the middleware tests have, where there's a `RunIntegrationTest` sort of method with parameters.